curvefs/src/tools/query/curvefs_copyset_query.cpp

```diff
@@ -0,0 +1,179 @@
+                    curvefs::mds::topology::TopoStatusCode::TOPO_OK) {
+                    // some error in copyset
+                    std::cout << "query copyset [ " << i.DebugString()
+                              << " ] from mds error." << std::endl;
+                    continue;
+                }
+                avaliableCopysetId.insert(i.copysetinfo().copysetid());
+
+                tmpType tmp;
+                tmp.set_copysetid(i.copysetinfo().copysetid());
+                tmp.set_poolid(i.copysetinfo().poolid());
+                for (auto const& j : i.copysetinfo().peers()) {
+                    tmp.set_allocated_peer(new curvefs::common::Peer(j));
+                    std::string addr;
+                    if (!SplitPeerId(j.address(), &addr)) {
+                        std::cerr << "copyset[" << tmp.copysetid()
+                                  << "] has error peerid: " << j.address()
+                                  << std::endl;
+                        break;
+                    }
+                    auto& queueRequest = ip2requests[addr];
+                    if (queueRequest.empty()) {
+                        queueRequest.push(curvefs::metaserver::copyset::
+                                              CopysetsStatusRequest());
+                    }
+                    *queueRequest.front().add_copysets() = tmp;
+                }
+            }
+            for (auto const& i : ip2requests) {
+                // set host
+                FLAGS_metaserverAddr = i.first;
+                status::CopysetStatusTool copysetStatustool("", false);
+                copysetStatustool.Init();
+                auto copysets = i.second.front().copysets();
+                copysetStatustool.SetRequestQueue(i.second);
+                ret = copysetStatustool.RunCommand();
+                auto copysetsStatus = copysetStatustool.GetResponse()->status();
+                for (int m = 0, n = 0;
+                     m < copysets.size() && n < copysetsStatus.size();
+                     ++m, ++n) {
+                    id2Status_[copysets[m].copysetid()].push_back(
+                        copysetsStatus[n]);
+                }
+            }
+            if (show_) {
+                for (auto const& i : avaliableCopysetId) {
+                    std::cout << "copyset: " << i << " status:[ ";
+                    for (auto const& j : id2Status_[i]) {
+                        std::cout << j.DebugString() << " ";
+                    }
+                    std::cout << "]." << std::endl;
+                }
+            }
+        } else if (show_) {
+            for (auto const& i : response_->copysetsinfo()) {
+                std::cout << i.DebugString() << std::endl;
+            }
+        }
+    }
+    return ret;
+}
+
+std::map<uint32_t, std::vector<StatusType>> CopysetQueryTool::GetId2Status() {
+    return id2Status_;
+}
+
+bool CopysetQueryTool::SplitPeerId(const std::string& peerId,
```

Comment:
why copy this function? put previous function into common, and directly use it

Reply:
> why copy this function? put previous function into common, and directly use it

fix